backend/app/Repositories/Elasticsearch/VisitorsFlow/ElasticsearchVisitorFlowBrowserRepository.php

```diff
@@ -0,0 +1,146 @@
+<?php
+declare(strict_types=1);
+
+namespace App\Repositories\Elasticsearch\VisitorsFlow;
+
+use App\Aggregates\VisitorsFlow\Aggregate;
+use App\Aggregates\VisitorsFlow\BrowserAggregate;
+use App\DataTransformer\VisitorsFlow\ParameterFlowCollection;
+use App\DataTransformer\VisitorsFlow\ParametersCollection;
+use App\Repositories\Elasticsearch\VisitorsFlow\Contracts\Criteria;
+use App\Repositories\Elasticsearch\VisitorsFlow\Contracts\VisitorFlowBrowserRepository;
+use Cviebrock\LaravelElasticsearch\Manager as ElasticsearchManager;
+
+class ElasticsearchVisitorFlowBrowserRepository implements VisitorFlowBrowserRepository
+{
+    const INDEX_NAME = 'browser-visitors-flow-index';
+    private $client;
+
+    public function __construct(ElasticsearchManager $client)
+    {
+        $this->client = $client;
+    }
+
+    public function save(Aggregate $browserAggregate): Aggregate
+    {
+        $this->client->index([
+            'index' => self::INDEX_NAME,
+            'id' => $browserAggregate->getId(),
+            'type' => '_doc',
+            'body' => $browserAggregate->toArray()
+        ]);
+
+        return $browserAggregate;
+    }
+
+    public function update(Aggregate $browserAggregate): Aggregate
+    {
+        $this->client->index([
+            'index' => self::INDEX_NAME,
+            'id' => $browserAggregate->getId(),
+            'type' => '_doc',
+            'body' => $browserAggregate->toArray()
+        ]);
+
+        return $browserAggregate;
+    }
+
+
+    public function getByCriteria(Criteria $criteria): ?BrowserAggregate
+    {
+        $params = [
+            'index' => self::INDEX_NAME,
+            'body' => [
+                'query' => [
+                    'bool' => [
```

Comment:
your criteria should build this array, it looks like Criteria doesn't have any sense in this context